assets/js/blocks/products/all-products/edit.tsx

```diff
@@ -161,46 +162,47 @@ class Editor extends Component {
 		);
 	};
 
-	renderEditMode = () => {
+	const renderEditMode = () => {
 		const onDone = () => {
-			const { block, setAttributes } = this.props;
 			setAttributes( {
-				layoutConfig: getProductLayoutConfig( block.innerBlocks ),
+				layoutConfig: getProductLayoutConfig( innerBlocks ),
 			} );
-			this.setState( { innerBlocks: block.innerBlocks } );
-			this.togglePreview();
+			setInnerBlocks( innerBlocks );
+			togglePreview();
 		};
 
 		const onCancel = () => {
-			const { block, replaceInnerBlocks } = this.props;
-			const { innerBlocks } = this.state;
-			replaceInnerBlocks( block.clientId, innerBlocks, false );
-			this.togglePreview();
+			replaceInnerBlocks( clientId, innerBlocks, false );
+			togglePreview();
 		};
 
 		const onReset = () => {
-			const { block, replaceInnerBlocks } = this.props;
-			const newBlocks = [];
-			DEFAULT_PRODUCT_LIST_LAYOUT.map( ( [ name, attributes ] ) => {
-				newBlocks.push( createBlock( name, attributes ) );
+			const newBlocks: BlockInstance[] = [];
+			DEFAULT_PRODUCT_LIST_LAYOUT.map( ( [ name, blockAttributes ] ) => {
+				newBlocks.push( createBlock( name, blockAttributes ) );
 				return true;
 			} );
-			replaceInnerBlocks( block.clientId, newBlocks, false );
-			this.setState( { innerBlocks: block.innerBlocks } );
+			replaceInnerBlocks( clientId, newBlocks, false );
+			setInnerBlocks( innerBlocks );
 		};
 
-		const InnerBlockProps = {
-			template: this.props.attributes.layoutConfig,
+		const InnerBlockProps: {
+			template: [ string, object? ][];
+			templateLock: boolean;
+			allowedBlocks: Array< string >;
+			renderAppender?: undefined | boolean;
```

Comment:
Considering the property is optional, it can be implicitly `undefined` hence you can just define it as optional `boolean`:

```suggestion
			renderAppender?: boolean;
```